### PR TITLE
fix(models): preserve CUDA context for MACE cuEq conversion

### PR DESCRIPTION
**Pull request by zubatyuk**
# ALCHEMI Toolkit Pull Request

## Description

Fix MACE cuEquivariance checkpoint loading so CUDA device selection and fused cuEq conversion use the device contract expected by mace-torch==0.3.15. The MACE converter enables its fused cuEq operation only when it receives the literal device string `"cuda"`; passing `torch.device("cuda")` or `torch.device("cuda:1")` can silently skip that fused path. This change keeps explicit CUDA indices active through `torch.cuda.device(...)` while passing `"cuda"` to the converter.

## Type of Change

- [x] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Performance improvement
- [ ] Documentation update
- [ ] Refactoring (no functional changes)
- [ ] CI/CD or infrastructure change

## Related Issues

None.

## Changes Made

- Normalize checkpoint target devices once and use the normalized `torch.device` for checkpoint loading and final model placement.
- Preserve explicit CUDA device indices during cuEquivariance conversion while passing `"cuda"` to the MACE converter so the fused cuEq path is selected.
- Warn and skip cuEquivariance conversion when `enable_cueq=True` is requested for a non-CUDA device, then continue loading the model on the requested device.

## Testing

- [x] Unit tests pass locally (`make pytest`)
- [x] Linting passes (`make lint`)
- [x] New tests added for new functionality meets coverage expectations?

Targeted MACE wrapper tests passed locally, including no-network tests for CPU warning behavior, device normalization, and explicit CUDA-context conversion, plus slow CUDA cuEq checkpoint tests.

## Checklist

- [x] I have read and understand the [Contributing Guidelines](../CONTRIBUTING.md)
- [ ] I have updated the [CHANGELOG.md](../CHANGELOG.md)
- [x] I have performed a self-review of my code
- [ ] I have added docstrings to new functions/classes
- [ ] I have updated the documentation (if applicable)

## Additional Notes

The relevant mace-torch==0.3.15 behavior is that cuEquivariance conversion decides whether to use fused convolution from an exact `device == "cuda"` check. The wrapper now keeps the requested CUDA device active with PyTorch's CUDA device context, which avoids losing explicit device selection while preventing the fused operation from being skipped by a non-string device argument.

> [!TIP]
> This repository uses Greptile, an AI code review service, to help conduct
> pull request reviews. We encourage contributors to read and consider suggestions
> made by Greptile, but note that human maintainers will provide the necessary
> reviews for merging: Greptile's comments are **not** a qualitative judgement
> of your code, nor is it an indication that the PR will be accepted/rejected.
> We encourage the use of emoji reactions to Greptile comments, depending on
> their usefulness and accuracy.

### CODE REVIEW

File: nvalchemi/models/mace.py [hunk 1]
```diff
@@ -520,9 +526,21 @@ def from_checkpoint(
...
 
-            model = _convert_mace_weights(model, return_model=True, device=device)
-
-        model = model.to(device)
+            if target_device.type != "cuda":
+                warnings.warn(
+                    "nvalchemi Toolkit MACE cuEquivariance conversion requires "
+                    "a CUDA device; skipping cuEquivariance conversion.",
+                    UserWarning,
+                )
```
[the remark this answers is not in the document]
Comment (zubatyuk) [reply nvalchemi/models/mace.py]: Fixed in 6b55f58a23dfe945d42115593e55dec12d46874d

File: nvalchemi/models/mace.py [hunk 2]
```diff
@@ -520,9 +526,22 @@ def from_checkpoint(
                 )
             from mace.cli.convert_e3nn_cueq import run as _convert_mace_weights
 
-            model = _convert_mace_weights(model, return_model=True, device=device)
-
-        model = model.to(device)
+            if target_device.type != "cuda":
```
Comment (laserkelvin) [inline nvalchemi/models/mace.py:526]: I'm wondering whether we should just throw an exception, instead of issuing a warning.

If the user wants to use CPU then they should disable cuEq. Conversely, if they want to run with cuEq, then they should make sure to configure the device correctly.

File: nvalchemi/models/mace.py [hunk 3]
```diff
@@ -520,9 +526,22 @@ def from_checkpoint(
...
+                warnings.warn(
+                    "nvalchemi Toolkit MACE cuEquivariance conversion requires "
+                    "a CUDA device; skipping cuEquivariance conversion.",
+                    UserWarning,
+                    stacklevel=2,
+                )
+            else:
+                with torch.cuda.device(target_device):
```
Comment (laserkelvin) [inline nvalchemi/models/mace.py]: What would the situation be where you would want to load the weights with cuEq and then move the weights to CPU?

Comment (zubatyuk) [reply nvalchemi/models/mace.py]: User can save state_dict. Cueq -> native model conversion is out of scope.

File: nvalchemi/models/mace.py (hunk 2 above)
Comment (zubatyuk) [reply nvalchemi/models/mace.py:526]: Fixed in bf3653e9ec13afd4cacea82118b43b4f896c71e6.

**Review by laserkelvin — APPROVED** [review]

**Comment by laserkelvin** [comment]
/ok to test [26f27bd](https://github.com/NVIDIA/nvalchemi-toolkit/pull/114/commits/26f27bdc43ea5a2c955f9a35e2fb8bbc63629b4b)
